jscomp/test/UntaggedVariants.js

```diff
@@ -176,6 +176,44 @@ var WithArray = {
 };
 
 function classify$6(x) {
+  switch (typeof x) {
+    case "string" :
+        return "string";
+    case "number" :
+        return "int";
+    case "boolean" :
+        if (x) {
+          return "true";
+        } else {
+          return "boolean";
+        }
+    case "object" :
+        return "Object" + x.name;
+    
+  }
+}
+
+var WithBool = {
+  classify: classify$6
+};
+
+function classify$7(x) {
+  if (x === false || x === true) {
+    if (x === false) {
+      return "false";
+    } else {
+      return "true";
+    }
+  } else {
```

Comment:
See comment in the thread that this type declaration should give an error (of overlap between bool and one bool constant) to avoid surprises like this one where the branch is dead code.